lib/authenticate/api.go

```diff
@@ -0,0 +1,10 @@
+package authenticate
+
+// Authenticator is an interface type that defines how to authenticate a user
+// with a username and password.
+type Authenticator interface {
+	// Authenticate will authenticate a user using the provided username and
+	// password. It returns true if the user is authenticated, else false (due
+	// to either invalid username or incorrect password).
+	Authenticate(username, password string) bool
```

Comment:
This inteface is problematic:
1. Passwords should be []byte
2. We should always be able to return errors, as the implementations might find errors such as network/other unknown conditions.

Also the name is problematic... I would call it passwordAuthenticate or similar